pkg/operator/ceph/nfs/nfs.go

```diff
@@ -55,7 +55,8 @@ type daemonConfig struct {
 
 // Create the ganesha server
 func (r *ReconcileCephNFS) upCephNFS(n *cephv1.CephNFS) error {
-	nfsToSkipReconcile, err := controller.GetDaemonsToSkipReconcile(r.clusterInfo.Context, r.context, n.Namespace, config.NfsType, AppName)
+	instanceLabelKey := "instance"
+	nfsToSkipReconcile, err := controller.GetDaemonsToSkipReconcile(r.clusterInfo.Context, r.context, n.Namespace, instanceLabelKey, AppName)
```

Comment:
If we change `"nfs"` to `"instance"`, then the `GetDaemonsToSkipReconcile()` command is going to log `found a "instance" pod to skip reconcile` which is not a helpful log.

As-is, it will log `found a "nfs" pod to skip reconcile` which is mostly helpful. Admittedly, I think `a` should be quoted with `%q` rather than `%s`, but that is an issue in `GetDaemonsToSkipReconcile()` and not with selecting the NFS pods.

Reply:
Probably we should include the deployment name in the log message, to clarify in this case since we need to use the "instance" label instead of a label with the daemon name.

Reply:
However, I do think the implementation of `GetDaemonsToSkipReconcile()` is naive and incomplete. It looks for reconciles to skip based on only 2 fields, `app=AppName` and an arbitrary input `label`, which it uses to determine the daemon IDs. However, that doesn't allow for distinction between multiple CRs in the same namespace when IDs are reused. For example, 2 CephNFSes would both have `instance=a`, so the method really should be fixed at some point to make sure it doesn't rely on each CR kind to only have one instance.

Reply:
I would suggest fixing line 66/67 instead. Instead of `if nfsToSkipReconcile.Has(id) {` use, `if nfsToSkipReconcile.Has(n.Name + "-" + id)`, which is a one-time fix, plus probably an update to unit tests. And it has the benefit of not making the assumption that there is only a single CephNFS present in the namespace. I am aware of at least I think two upstream users who have reported having multiple CephNFSes in their clusters.

Reply:
Good point about that query being incorrect with multiple nfs'es, so sounds perfect to adjust the `Has()` statement instead of changing the label.

Reply:
Yaa correct we should fix the has, mine assumption was each nfs cr pod will get unique id and can have only one instance seems wrong.